lib/rest/RequestHandler.js

```diff
@@ -110,21 +111,72 @@ class RequestHandler {
                     if(file) {
                         if(Array.isArray(file)) {
                             data = new MultipartData();
+                            const attachments = [];
+
                             headers["Content-Type"] = "multipart/form-data; boundary=" + data.boundary;
-                            file.forEach(function(f) {
+                            file.forEach(function(f,i) {
                                 if(!f.file) {
                                     return;
                                 }
-                                data.attach(f.name, f.file, f.name);
+
+                                attachments.push({
+                                    id: i,
+                                    description: f.description,
+                                    filename: f.name
+                                });
+
+                                data.attach(`files[${i}]`, f.file, f.name);
+
                             });
+
                             if(body) {
-                                data.attach("payload_json", body);
+                                if(typeof body === "string"){
+                                    // eslint-disable-next-line object-shorthand
+                                    body = {
+                                        content: body,
+                                        attachments
+                                    };
```

Comment:
Instead of disabling the eslint rule, you should follow the styling
```suggestion
                                        body = {
                                            content: body,
                                            attachments: attachments
                                        };
```